src/views/Drive/components/DriveExplorer/components/DriveItemContextMenu.tsx

```diff
@@ -93,14 +94,46 @@ const getDownloadMenuItem = (downloadItems: (target?) => void) => ({
   },
 });
 
-const getVersionHistoryMenuItem = (viewVersionHistory: (target?) => void) => ({
-  name: t('drive.dropdown.versionHistory'),
-  icon: ClockCounterClockwise,
-  action: viewVersionHistory,
-  disabled: (item) => {
-    return item.isFolder;
-  },
-});
+export type VersionHistoryMenuConfig = {
+  isLocked: boolean;
+  isExtensionAllowed: boolean;
+  onUpgradeClick?: () => void;
+};
+
+const getVersionHistoryMenuItem = (
+  viewVersionHistory: (target?) => void,
+  config?: VersionHistoryMenuConfig,
+): MenuItemType<DriveItemData> => {
+  const isLocked = config?.isLocked ?? false;
+  const isExtensionAllowed = config?.isExtensionAllowed ?? true;
+  const onUpgradeClick = config?.onUpgradeClick;
+
+  if (isLocked) {
+    return {
+      name: t('drive.dropdown.versionHistory') as string,
+      icon: LockSimple,
+      action: onUpgradeClick,
+      disabled: () => false,
+      node: (
+        <div
+          className="flex flex-row items-center space-x-2"
+          data-locked="true"
+          style={{ opacity: 0.5, cursor: 'default' }}
+        >
+          <LockSimple size={20} />
+          <span>{t('drive.dropdown.versionHistory')}</span>
+        </div>
+      ),
+    };
```

Comment:
why is this component necesary? with only passing `icon` field is not displayed correctly?